functions/Get-DbaJobOutputFile.ps1

```diff
@@ -0,0 +1,201 @@
+            $row.Instance = $Server.Instancename
+            $row.Job = $Job.Name
+            $row.JobStep = $step.Name
+            $row.FileName = $FileName
+          
+            #Add new row to table
+            $table.Rows.Add($row)
+        }
+         
+    }
+    PROCESS
+    {
+        foreach ($instance in $sqlserver)
+        {
+            $Server = Connect-SqlServer -SqlServer $instance  -SqlCredential $SqlCredential
+            $jobs = $Server.JobServer.Jobs
+            if ($JobName)
+            {
+                $Job = $server.JobServer.Jobs[$JobName]
+                $Servername = $Server.ComputerNamePhysicalNetBIOS
+                foreach($Step in $Job.JobSteps)
+                {
+                $fileName = $Step.OutputFileName
+                if($fileName -eq '')
+                {
+                    Add-Row 
+                }
+                else
+                {         
+                    if($FileName.StartsWith('\\') -eq $false)
+            {
+                $fileName = '\\' + $Servername + '\' + $Filename.Replace(':','$')
+                Add-Row 
+            }
+            else
+            {
+              Add-Row 
+            }
+            if($OpenFile)
+            {
+                if(Test-Path $fileName)
+                {
+                    If ($Pscmdlet.ShouldProcess("$FileName", "Opening File $FileName"))
+				    {
+                        Invoke-Item $fileName
+                    }
+                }
+                else
+                {
+                    Write-Output 'No File to open'
+                }
+            }
+        }
+            }
+            }
+            else
+            {
+                foreach($Job in $Jobs)
+                {
+            $Servername = $Server.ComputerNamePhysicalNetBIOS
+            foreach($Step in $Job.JobSteps)
+            {
+                $fileName = $Step.OutputFileName
+                if($fileName -eq '')
+                {
+                    Add-Row 
+                }
+                else
+                {        
+                    if($FileName.StartsWith('\\') -eq $false)
+                    {
+                $fileName = '\\' + $Servername + '\' + $Filename.Replace(':','$')
+                Add-Row 
+            }
+                    else
+                    {
+                Add-Row 
+            }
+                    if($OpenFile)
+                    {
+                if(Test-Path $fileName)
+                {
+                    If ($Pscmdlet.ShouldProcess("$FileName", "Opening File $FileName"))
+				    {
+                        Invoke-Item $fileName
+                    }
+                }
+                else
+                {
+                    Write-Output 'No File to open'
+                }
+            }
+                }
+            }
+        }
+            }             
+            $server.ConnectionContext.Disconnect()
+        }
+    }
+    END
+    {
+        $table  
```

Comment:
Please ensure the output is not held until the END -- it should be done in the pipeline. I started that, my bad!